src/ILCompiler.CppCodeGen/src/CppCodeGen/CppWriter.cs

```diff
@@ -1217,6 +1239,37 @@ private void OutputExternCSignatures()
             Out.Write(sb.ToString());
         }
 
+        /// <summary>
+        /// Output C++ code for a given unboxingStubNode
+        /// </summary>
+        /// <param name="unboxingStubNode">The unboxing node to be output</param>
+        /// <param name="methodImplementations">The buffer in which to write out the C++ code</param>
+        private void OutputUnboxNode(UnboxingStubNode unboxingStubNode)
```

Comment:
Would it be possible to just allocate a normal `CppMethodCodeNode` for these, where `_methodCode` is this code and `_dependencies` points to the `CppMethodCodeNode` of the target?

It would fix the TODO here: https://github.com/dotnet/corert/blob/e391acfbb7fb440f4b5469623f33a4b5d23be369/src/ILCompiler.CppCodeGen/src/Compiler/DependencyAnalysis/CppCodegenNodeFactory.cs#L35

It feels odd to allocate the `UnboxingStubNode` (which represents a machine code helper) and then try to patch things up later.